### ISSUE TITLE
SOLD2 samples line descriptors with a corner-aligned grid under align_corners=False, drifting up to one image pixel from the reference implementation

### ISSUE THREAD

**Issue by VanniLeonardo** [kornia/kornia#4554]
### What happened?

kornia's SOLD2 port samples line descriptors at slightly different positions from the reference implementation the pretrained weights were trained with, and the gap grows toward the bottom-right of the image.

`WunschLineMatcher.forward` builds its sampling grid with `keypoints_to_grid` (`kornia/feature/sold2/sold2.py:331`), which normalizes through `normalize_pixel_coordinates`, i.e. the corner-aligned mapping `2p / (size - 1) - 1`. It then reads the descriptor map with `F.grid_sample(..., align_corners=False)`, which interprets `[-1, 1]` as the outer pixel edges. The reference implementation ([cvg/SOLD2 `sold2/misc/geometry_utils.py`](https://github.com/cvg/SOLD2/blob/main/sold2/misc/geometry_utils.py)) normalizes with `keypoints * 2 / img_size - 1` and also samples at `grid_sample`'s default `align_corners=False`, so its grid and its sampler use the same convention.

The two normalizations differ by `2p / (W (W - 1))` in normalized units, which is `p / (W - 1)` image pixels: zero at the origin and one full image pixel at the far edge, independent of the descriptor stride. It has been this way since the port landed in #1844.

Measured on a 480x640 image with the default `grid_size=8`, so a 60x80 descriptor map. Columns are the descriptor-map coordinate that `grid_sample` actually reads:

| keypoint `(i, j)` | kornia reads | upstream reads | difference, desc px | difference, image px |
|---|---|---|---|---|
| (0, 0) | (-0.500, -0.500) | (-0.500, -0.500) | 0 | 0 |
| (240, 320) | (39.563, 29.563) | (39.500, 29.500) | +0.063 | +0.50 |
| (479, 639) | (79.500, 59.500) | (79.375, 59.375) | +0.125 | +1.00 |
| (100, 639) | (79.500, 12.026) | (79.375, 12.000) | +0.125 / +0.026 | +1.00 / +0.21 |

At the far edge kornia's sample sits exactly half a descriptor pixel past the last centre, so half its bilinear weight lands in zero padding while upstream's does not. On a unit-norm random descriptor map the sampled norms at `(479, 639)`, `(100, 639)` and `(479, 300)` come out `0.250`, `0.486`, `0.468` for kornia against `0.391`, `0.625`, `0.625` for upstream. `F.normalize` afterwards undoes the scale, so what survives is the shift itself and a different blend of neighbouring descriptors along the right and bottom borders.

The magnitude is small, at most one image pixel and one eighth of a descriptor pixel, so this is a fidelity deviation from the reference weights rather than a gross failure. It is the same corner-aligned-versus-edge mismatch as #3904, on a code path #3945 did not touch.

### Minimal reproduction

```python
import torch
from kornia.feature.sold2.sold2 import keypoints_to_grid

def upstream_grid(kp, img_size):  # cvg/SOLD2, sold2/misc/geometry_utils.py
    n = kp.shape[-2]
    g = kp.float() * 2.0 / torch.tensor(img_size, dtype=torch.float32) - 1.0
    return g[..., [1, 0]].view(-1, n, 1, 2)

H, W, g = 480, 640, 8
hd, wd = H // g, W // g
kp = torch.tensor([[0.0, 0.0], [240.0, 320.0], [479.0, 639.0]])          # ij, image pixels

def where_grid_sample_reads(grid):                                     # align_corners=False inverse
    return (grid[0, :, 0, :] + 1) * torch.tensor([wd, hd]) / 2 - 0.5

print(where_grid_sample_reads(keypoints_to_grid(kp, (H, W))))
print(where_grid_sample_reads(upstream_grid(kp, (H, W))))
```

Observed:

```
tensor([[-0.5000, -0.5000], [39.5625, 29.5625], [79.5000, 59.5000]])   # kornia
tensor([[-0.5000, -0.5000], [39.5000, 29.5000], [79.3750, 59.3750]])   # upstream
```

Expected: the two agree, since the weights were trained with the second.

### Environment

- Kornia version: 0.9.0rc1 (current `main`, `1590df48`)
- PyTorch version: 2.9.1+cu128
- Python version: 3.11.14
- OS / device: Linux, CPU

### Anything else?

The fix is one line and self-contained: have `keypoints_to_grid` compute `keypoints * 2 / img_size - 1` as upstream does, instead of calling `normalize_pixel_coordinates`. It should not wait for #4504, and it should not use the `align_corners` parameter that issue proposes either. The half-pixel mapping `(2p + 1) / W - 1` is not what upstream does; it differs from upstream's formula by a constant half image pixel, and the target here is the convention the weights were trained under, not the pixel-centre convention. `keypoints_to_grid` is a module-level helper, not exported and not tested, so this is not a public API change.

Upstream also samples line scores and descriptors in two further places (`line_matching.py` lines 163 to 174 and 246 to 249). kornia's port has only the one site, so nothing else inherits this.

Found while surveying the callers of `normalize_pixel_coordinates` for #4504. Investigated with Claude Code; every figure above is from an executed run on current `main`.

**Comment by VanniLeonardo** [comment]
I will take this one. It is a one-line change in `keypoints_to_grid` plus a regression test against the upstream formula, and it does not depend on #4504. A maintainer will need to assign me.
